toolkit/tools/srpmpacker/srpmpacker.go

```diff
@@ -605,6 +611,8 @@ func packSRPMs(specStates []*specState, distTag, buildDir string, templateSrcCon
 			logger.Log.Errorf("Failed to pack (%s). Error: %s", result.specFile, result.err)
 			err = result.err
 			close(cancel)
+			// Drain the semaphore to ensure all workers can exit quickly
+			drainChannel(netOpsSemaphore)
```

Comment:
This feels a tad messy. And it isn't immediately clear how or why this works.

Reply:
Piped the cancel channel all the way through.